src/test/ui/consts/offset_ub.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(const_ptr_offset)]
+use std::ptr;
+
+// normalize-stderr-test "alloc\d+" -> "allocN"
+
+pub const BEFORE_START: *const u8 = unsafe { (&0u8 as *const u8).offset(-1) };
+pub const AFTER_END: *const u8 = unsafe { (&0u8 as *const u8).offset(2) };
+pub const AFTER_ARRAY: *const u8 = unsafe { [0u8; 100].as_ptr().offset(101) };
+
+pub const OVERFLOW: *const u16 = unsafe { [0u16; 1].as_ptr().offset(isize::MAX) };
+pub const UNDERFLOW: *const u16 = unsafe { [0u16; 1].as_ptr().offset(isize::MIN) };
+pub const OVERFLOW_ADDRESS_SPACE: *const u8 = unsafe { (usize::MAX as *const u8).offset(2) };
+pub const UNDERFLOW_ADDRESS_SPACE: *const u8 = unsafe { (1 as *const u8).offset(-2) };
+
+pub const ZERO_SIZED_ALLOC: *const u8 = unsafe { [0u8; 0].as_ptr().offset(1) };
+pub const DANGLING: *const u8 = unsafe { ptr::NonNull::<u8>::dangling().as_ptr().offset(4) };
```

Comment:
Please add `ERROR` annotations here to make sure this test cannot pass when these constants do not error.

Reply:
Note that if the test passes, this will fail the test suite. You need a `// check-pass` comment or sth to make it not fail the test suite. Still, annotations are better so you see something directly in the test

Reply:
That just checks that *any one* of these constants raises an error. We should make sure *all* of them do.

Reply:
I'm in full agreement. That's what I meant by `Still, annotations are better so you see something directly in the test`

Reply:
This makes sense, I thought that for UI tests we don't need the `ERROR` annotations (because we check that the stderr matches exactly). But it looks like other tests do the same thing.

Reply:
The stderr is auto-changed by `--bless`. I'd rather not rely on a reviewer realizing that a 1k line patch accidentally removed one of the errors here.

That's why we have both `ERROR` annotations and stderr files.